src/core/datastore/persisters/key-value-persister.js

```diff
@@ -0,0 +1,67 @@
+const _cache = {};
+
+export class KeyValuePersister {
+  /** @type {Boolean} */
+  _cacheEnabled;
+  /** @type {Number} */
+  _ttl;
+
+  // TODO: implement TTL, make _cache a constructor argument?
+  constructor(cacheEnabled = true, ttl = Infinity) {
+    this._cacheEnabled = cacheEnabled;
+    this._ttl = ttl;
+  }
+
+  readEntities(key) {
+    if (this._cacheEnabled && _cache[key]) {
+      return Promise.resolve(_cache[key]);
+    }
+
+    return this._readFromPersistance(key)
+      .then((entities) => {
+        entities = entities || [];
+        if (this._cacheEnabled) {
+          _cache[key] = entities;
+        }
+        return entities;
+      });
+  }
+
+  persistEntities(key, entities) {
+    if (this._cacheEnabled) {
+      delete _cache[key];
+    }
+    return this._writeToPersistance(key, entities)
+      .then((result) => {
+        if (this._cacheEnabled && this._ttl < Infinity) {
```

Comment:
Is `ttl` implemented? I thought from the TODO above that it was not.

Reply:
Well, the class supports it, but nobody is using it. There is a general decision to be made about it, so the todo is there to make sure I don't forget it. There is a brief discussion about it in the previous "set" of PRs. It's still outstanding as to whether to keep it and in what form.

Reply:
My comment from above, about this TTL, might have been more suitable for over here :)